symphony-bdk-core/src/main/java/com/symphony/bdk/core/SymphonyBdkBuilder.java

```diff
@@ -0,0 +1,153 @@
+package com.symphony.bdk.core;
+
+import com.symphony.bdk.core.auth.AuthenticatorFactory;
+import com.symphony.bdk.core.auth.exception.AuthInitializationException;
+import com.symphony.bdk.core.auth.exception.AuthUnauthorizedException;
+import com.symphony.bdk.core.client.ApiClientFactory;
+import com.symphony.bdk.core.config.BdkConfigLoader;
+import com.symphony.bdk.core.config.exception.BdkConfigException;
+import com.symphony.bdk.core.config.model.BdkConfig;
+
+import com.symphony.bdk.core.util.ServiceLookup;
+import com.symphony.bdk.http.api.ApiClientBuilderProvider;
+
+import lombok.Generated;
+import org.apiguardian.api.API;
+
+import java.io.InputStream;
+
+import javax.annotation.Nonnull;
+import javax.annotation.Nullable;
+
+/**
+ * Fluent builder for advanced configuration of the {@link SymphonyBdk} entry point.
+ */
+@Generated // a bit tricky to get acceptable coverage
+@API(status = API.Status.STABLE)
+public class SymphonyBdkBuilder {
+
+  private BdkConfig config;
+  private ApiClientBuilderProvider apiClientBuilderProvider;
+  private AuthenticatorFactory authenticatorFactory;
+  private ApiClientFactory apiClientFactory;
+
+  /**
+   * With {@link BdkConfig} POJO.
+   *
+   * @param config config POJO.
+   * @return updated builder.
+   */
+  public SymphonyBdkBuilder config(@Nonnull BdkConfig config) {
+    this.config = config;
+    return this;
+  }
+
+  /**
+   * With config from Symphony directory.
+   *
+   * @param relPath relative configuration file location from your ${user.home}/.symphony directory.
+   * @return updated builder.
+   * @throws BdkConfigException config file loading issue.
+   */
+  public SymphonyBdkBuilder configFromSymphonyDir(@Nonnull String relPath) throws BdkConfigException {
+    this.config = BdkConfigLoader.loadFromSymphonyDir(relPath);
+    return this;
+  }
+
+  /**
+   * With config from classpath.
+   *
+   * @param configPath relative classpath location of the configuration file.
+   * @return updated builder.
+   * @throws BdkConfigException config file loading issue.
+   */
+  public SymphonyBdkBuilder configFromClasspath(@Nonnull String configPath) throws BdkConfigException {
```

Comment:
isn't enough to just expose config() and let the caller build the config himself?

Reply:
I'm ok with that, I'll just keep the `SymphonyBdkBuilder#config(BdkConfig)`